Create REFERENCE.md (#394)
astronomer/astro-sdk: REFERENCE.md

    ```

3. Set the following environment variable so that `astro` can pass table objects between tasks:

    ```shell script
    AIRFLOW__CORE__ENABLE_XCOM_PICKLING=True
    ```

# Usage

## Setting Input and Output Tables

Before we can complete any transformations, we need to define a way to get our tables in and out of Airflow. We can do this by defining either `Table` or `TempTable` objects in the `input_table` and `output_table` parameters of our table instantiations.

### The Table class

To instantiate a table or bring in a table from a database into the `astro-sdk-python` ecosystem, you can pass a `Table` object into the class. This `Table` object will contain all of the metadata that's necessary for handling table creation between tasks. After you define a Table's metadata in the beginning of your pipeline, `astro-sdk-python` can automatically pass that metadata along to downstream tasks.

In the following example, we define our table in the DAG instantiation. In each subsequent task, we pass in only an input `Table` argument because `astro-sdk-python` automatically passes in the additional context from our original `input_table` parameter.

```python

> from astro-sdk-python import sql as aql
> from astro-sdk-python.sql.table import Table

Reviewer: ```suggestion
from astro import sql as aql
from astro.sql.table import Table
```
```suggestion
from astro-sdk-python import sql as aql
from astro-sdk-python.sql.table import Table
```

> from astro-sdk-python import sql as aql
> from astro-sdk-python.sql.table import Table

Reviewer: ```suggestion
from astro import sql as aql
from astro.sql.table import Table
```
```suggestion
from astro-sdk-python import sql as aql
from astro-sdk-python.sql.table import Table
```

    return "SELECT * FROM {{input_table_2}}"

with dag:
    my_table = my_first_sql_transformation(
        input_table=Table(table_name="foo", database="bar", conn_id="postgres_conn")
    )
    my_second_sql_transformation(my_table)
```

### The TempTable Class

Following the traditional dev ops concept of [pets vs. cattle](http://cloudscaling.com/blog/cloud-computing/the-history-of-pets-vs-cattle/), you can decide whether the result of a function is a "pet" (e.g. a named table that you would want to reference later), or a "cattle" that can be deleted at any time for garbage collection.

If you want to ensure that the output of your task is a cattle, you can declare it as a nameless `TempTable`. This places the output into your temp schema,
which can be later bulk deleted. By default, all `aql.transform` functions will output to TempTables unless a `Table` object is used in the `output_table`
argument.

In the following example DAG, we set an `output_table` to a nameless `TempTable` meaning that any output from this DAG will be deleted once the DAG completes. If we wanted to keep our output, we would simply update the parameter to instantiate a `Table` instead.

```python

> from astro-sdk-python import sql as aql
> from astro-sdk-python.sql.table import Table, TempTable

Reviewer: ```suggestion
from astro import sql as aql
from astro.sql.table import Table, TempTable
```
```suggestion
from astro-sdk-python import sql as aql
from astro-sdk-python.sql.table import Table, TempTable
```

> from astro-sdk-python import sql as aql
> from astro-sdk-python.sql.table import Table, TempTable

Reviewer: ```suggestion
from astro import sql as aql
from astro.sql.table import Table, TempTable
```
```suggestion
from astro-sdk-python import sql as aql
from astro-sdk-python.sql.table import Table, TempTable
```

@aql.transform
def my_first_sql_transformation(input_table: Table):
    return "SELECT * FROM {{input_table}}"

@aql.transform
def my_second_sql_transformation(input_table_2: Table):
    return "SELECT * FROM {{input_table_2}}"

with dag:
    my_table = my_first_sql_transformation(
        input_table=Table(table_name="foo", database="bar", conn_id="postgres_conn"),
        output_table=TempTable(database="bar", conn_id="postgres_conn"),
    )
    my_second_sql_transformation(my_table)
```

## Loading Data

To create an ELT pipeline, users can first load CSV or parquet data from either local, S3, or GCS into a SQL database with the `load_sql` function.
To interact with S3, you must set an S3 Airflow connection in the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`  environment variables.

In the following example, we load data from S3 by specifying the path and connection ID for our S3 database in `aql.load_file`:

```python

> from astro-sdk-python import sql as aql
> from astro-sdk-python.sql.table import Table

Reviewer: ```suggestion
from astro import sql as aql
from astro.sql.table import Table, TempTable
```
```suggestion
from astro-sdk-python import sql as aql
from astro-sdk-python.sql.table import Table
```
